app/javascript/Components/assignment_summary_table.jsx

```diff
@@ -30,16 +48,145 @@ export class AssignmentSummaryTable extends React.Component {
+              },
+            }}
+          >
+            {row.getIsExpanded() ? "▼" : "▶"}
+          </button>
+        ) : (
+          " "
+        );
+      },
     });
 
+    const fixedColumns = [
+      columnHelper.accessor("inactive", {
+        id: "inactive",
+      }),
+      columnHelper.accessor("group_name", {
+        id: "group_name",
+        header: () => I18n.t("activerecord.models.group.one"),
+        size: 100,
+        enableResizing: true,
+        cell: info => {
+          if (info.row.original.result_id) {
+            const path = Routes.edit_course_result_path(
+              this.props.course_id,
+              info.row.original.result_id
+            );
+            return (
+              <a href={path}>
+                {info.row.original.group_name}
+                {this.memberDisplay(info.row.original.group_name, info.row.original.members)}
+              </a>
+            );
+          } else {
+            return (
+              <span>
+                {info.row.original.group_name}
+                {this.memberDisplay(info.row.original.group_name, info.row.original.members)}
+              </span>
+            );
+          }
+        },
+      }),
+      markingStateColumn(this.state.marking_states, this.state.markingStateFilter),
+      columnHelper.accessor("tags", {
+        header: () => I18n.t("activerecord.models.tag.other"),
+        size: 90,
+        enableResizing: true,
+        cell: info => (
+          <ul className="tag-list">
+            {info.row.original.tags.map(tag => (
+              <li key={`${info.row.original._id}-${tag}`} className="tag-element">
+                {tag}
+              </li>
+            ))}
+          </ul>
+        ),
+        minWidth: 80,
+        enableSorting: false,
+        filterFn: (row, columnId, filterValue) => {
+          if (filterValue) {
+            // Check tag names
+            return row.original.tags.some(tag => tag.includes(filterValue));
+          } else {
+            return true;
+          }
+        },
+      }),
+      columnHelper.accessor("final_grade", {
+        header: () => I18n.t("results.total_mark"),
+        size: 100,
+        enableResizing: true,
+        cell: info => {
+          if (info.row.original.final_grade || info.row.original.final_grade === 0) {
+            const max_mark = Math.round(info.row.original.max_mark * 100) / 100;
+            return info.row.original.final_grade + " / " + max_mark;
+          } else {
+            return "";
+          }
+        },
+        meta: {className: "number"},
+        enableColumnFilter: false,
+        sortDescFirst: true,
+      }),
+    ];
+
+    const criteriaColumns = this.state.criteriaColumns.map(col =>
+      columnHelper.accessor(col.accessor, {
+        id: col.id,
+        header: () => col.Header,
+        size: 100,
+        enableResizing: true,
+        cell: value => value.getValue(),
+        meta: {className: "number"},
```

Comment:
This isn' exactly correct. Look at the values passed in the `criteriaColumns` state; there are additional attributes that are meant to be used here.

To see the difference in the UI, *assign graders to specific criteria (and students)* and then login as one of the graders and visit the table. Criteria that haven't been assigned should be greyed out due to the styling.